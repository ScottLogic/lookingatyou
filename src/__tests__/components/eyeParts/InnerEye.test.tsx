import { mount, shallow } from 'enzyme';
import React from 'react';
import {
    InnerEye,
    InnerEyeProps,
} from '../../../components/eye/eyeParts/InnerEye';

let props: InnerEyeProps;

const data = new Uint8ClampedArray(400);

for (let i = 0; i < 400; i += 4) {
    data[i + 0] = 190; // R value
    data[i + 1] = 0; // G value
    data[i + 2] = 210; // B value
    data[i + 3] = 255; // A value
}

const imageData = { data, width: 10, height: 10 };

describe('InnerEye', () => {
    beforeEach(() => {
        props = {
            irisRadius: 200,
            dilatedCoefficient: 1,
            innerX: 0,
            innerY: 0,
            irisColor: 'blue',
            fps: 30,
            height: 800,
            width: 1000,
            pupilColor: 'black',
            pupilRadius: 100,
            scleraRadius: 500,
            reflectionOpacity: 0.5,
            showReflection: true,
            target: { x: 0, y: 0 },
            innerPath: '',
            image: undefined,
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(<InnerEye {...props} />).debug();
        expect(wrapper).toMatchSnapshot();
    });
});
