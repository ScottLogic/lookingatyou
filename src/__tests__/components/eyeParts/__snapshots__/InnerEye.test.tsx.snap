// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InnerEye should render correctly 1`] = `
"<g className=\\"inner\\" style={{...}} transform=\\"\\\\n                    undefined \\\\n                    translate(0,0)\\\\n                \\">
  <circle className=\\"iris\\" r={200} fill=\\"url(#irisGradient)\\" />
  <path className=\\"irisStyling\\" d=\\"\\" fill=\\"#0000cc\\" />
  <g className=\\"pupil\\" style={{...}} transform=\\"scale(1)\\">
    <foreignObject width={200} height={200} x={-100} y={-100}>
      <canvas width={200} height={200} />
    </foreignObject>
    <circle className=\\"pupil\\" r={100} fill=\\"url(#pupilGradient)\\" stroke=\\"black\\" strokeWidth=\\"2\\" />
  </g>
  <ellipse className=\\"innerShine\\" rx={37.5} ry={75} fill=\\"url(#shineGradient)\\" transform=\\"skewX(30) translate(100,-50)\\" />
  <ellipse className=\\"outerShine\\" rx={50} ry={100} fill=\\"url(#shineGradient)\\" transform=\\"skewX(30) translate(200,-110.00000000000001)\\" />
</g>"
`;
